test/jasmine/tests/updatemenus_test.js

```diff
@@ -462,6 +462,52 @@ describe('update menus interactions', function() {
         });
     });
 
+    it('should update correctly on failed binding comparisons', function(done) {
```

Comment:
Ah, a full end to end test of the specific failing case. Thanks! 👍

To elaborate, the specific difficulty is knowing when to dive into array contents. It's a little unpleasant since… do you do a deep comparison? Limit the deep comparison to when there aren't 1000 elements?

To limit the intelligence (which correlates with bugginess), my solution was to only compare arrays if they're length 1. I think that's a reasonable compromise since overall there are quite a few tests here (though this case was missed, obv 😞 )